opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -600,12 +600,14 @@ namespace Opm
 
         // Operability checking is not free
         // Only check wells under BHP and THP control
-        if(bhp_controled || thp_controled) {
+        bool check_thp = thp_controled || this->operability_status_.thp_limit_violated_but_not_switched;
+        if (check_thp || bhp_controled) {
             updateIPR(ebos_simulator, deferred_logger);
             checkOperabilityUnderBHPLimit(well_state, ebos_simulator, deferred_logger);
         }
         // we do some extra checking for wells under THP control.
-        if (thp_controled) {
+        const auto& summaryState = ebos_simulator.vanguard().summaryState();
```

Comment:
This `summaryState` object isn't used anywhere.  Is it supposed to be?  If not, we should not introduce it.

Reply:
I did some final cleanup and forgot to remove it. Thanks for noticing. I will update the code accordingly.

Reply:
> I did some final cleanup and forgot to remove it.

Ah, I see.

> Thanks for noticing.

I had a lot of help from the [build check](https://ci.opm-project.org/job/opm-simulators-PR-builder/3540/gcc/).